ProjectGroupA/src/main/java/com/fundation/search/model/Search.java

```diff
@@ -43,27 +44,36 @@ public Search() {
 		String name = file.getFileName();
 		String ext = file.getExt();
 		String path = file.getPath();
-		String size = file.getSize();
-		String operator = file.getOperator();
-		String sizeScale = file.getSizeScale();
-		Boolean isDirectory = file.getIsDirectory();
-		
+				
 		String res = "dir ";
-		if (isDirectory){
-			res = res +  "\"" + path + "\" /ad /b /s";
-		} else if (!path.endsWith("\\")) {
-			res = res + "\"" + path + "\\" + name + "." + ext + "\"" + " /s /b /a-d";
+		
+		//If it is a directory, the readOnly parameter is not valid.
+		if ( file.getIsDirectory() ){
+			if ( !file.getIsHidden() ) 
+				res = res +  "\"" + path + "\" /s /b /ad";
+			else
+				res = res +  "\"" + path + "\" /s /b /adh";
+			
 		} else {
-			res = res +  "\"" + path + name + "." + ext + "\"" + " /s /b /a-d";
+			String parameter = "/a";
+			if (file.getIsHidden())
+				parameter = parameter + "h";
+			if (file.getIsReadOnly())
```

Comment:
if statements always use braces even when it only has one-line. please set braces {}

Reply:
done